src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -290,70 +295,105 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
     }
   }
 
-  const inputHandler = useCallback(
+  /**
+   * handleInputChange
+   *
+   * Determines the form's state from input by onChange event.
+   * Valid inputs:
+   * - Non-empty numeric values including zero
+   * Error states:
+   * - Insufficient funds - give > have
+   * - Empty amount - input = ''
+   * - Not enough native token - gas > have
+   */
+  const handleInputChange = useCallback(
     async (inputValue: string) => {
-      setLoading(true)
       setValue(SendFormFields.SendMax, false)
+
       const key =
         fieldName !== SendFormFields.FiatAmount
           ? SendFormFields.FiatAmount
           : SendFormFields.CryptoAmount
-      if (inputValue === '') {
-        // Don't show an error message when the input is empty
-        setValue(SendFormFields.AmountFieldError, '')
-        setLoading(false)
-        // Set value of the other input to an empty string as well
-        setValue(key, '')
-        return
-      }
-      const amount =
-        fieldName === SendFormFields.FiatAmount
-          ? bnOrZero(bn(inputValue).div(price)).toString()
-          : bnOrZero(bn(inputValue).times(price)).toString()
 
-      setValue(key, amount)
+      try {
+        if (inputValue === '') {
+          // TODO: use isDirty instead
```

Comment:
Can we `useDirty` here?

Reply:
I'm not clear as to why we would use isDirty here. This is within the handleInputChange function, so this code would not ever be hit unless isDirty === true. I'm not sure this TODO is valid.

Reply:
Yeah, being in the context of `handleInputChange` means that the field will inherently have a value different than the default one. We can remove this comment

Reply:
Removed